src/ensembl/utils/plugin.py

```diff
@@ -63,6 +64,23 @@ def pytest_addoption(parser: Parser) -> None:
     )
 
 
+def pytest_configure(config: Config) -> None:
+    """Allows plugins and conftest files to perform initial configuration.
+
+    More information: https://docs.pytest.org/en/latest/reference/reference.html#std-hook-pytest_configure
+
+    Args:
+        config: The pytest config object.
+
+    """
+    # Load server information
+    server_url = make_url(config.getoption("server"))
+    # If password set, treat it as an environment variable that needs to be resolved
+    if server_url.password:
+        server_url = server_url.set(password=os.path.expandvars(server_url.password))
+        config.option.server = server_url.render_as_string(hide_password=False)
```

Comment:
I'm mindful that most test setups don't need this config adjustment, since if they did, this hook would have been introduced sooner.

Should we perhaps skip it entirely if no `"server"` is configured?
```suggestion
    server = config.getoption("server")
    if server:
        server_url = make_url(server)
        # If password set, treat it as an environment variable that needs to be resolved
        if server_url.password:
            server_url = server_url.set(password=os.path.expandvars(server_url.password))
            config.option.server = server_url.render_as_string(hide_password=False)
```

Reply:
There is no need for this because the server config has a non-empty default value: `sqlite:///` [[link](https://github.com/Ensembl/ensembl-utils/blob/d31a44661bbce1fcd1e6b1a75a13ecc4d6c1a6f7/src/ensembl/utils/plugin.py#L55)].

Reply:
Fair enough.